src/Platform/Microsoft.Testing.Platform/OutputDevice/Terminal/TerminalTestReporter.cs

```diff
@@ -644,10 +648,65 @@ internal void AssemblyRunCompleted()
         _terminalWithProgress.RemoveWorker(assemblyRun.SlotIndex);
     }
 
-    private static string? NormalizeSpecialCharacters(string? text)
-        => text?.Replace('\0', '\x2400')
-            // escape char
-            .Replace('\x001b', '\x241b');
+    // SearchValues for efficient detection of control characters
+#if NET8_0_OR_GREATER
+    private static readonly SearchValues<char> AllControlChars = SearchValues.Create("\x0000\x0001\x0002\x0003\x0004\x0005\x0006\x0007\x0008\x0009\x000A\x000B\x000C\x000D\x000E\x000F\x0010\x0011\x0012\x0013\x0014\x0015\x0016\x0017\x0018\x0019\x001A\x001B\x001C\x001D\x001E\x001F");
+    private static readonly SearchValues<char> NonWhitespaceControlChars = SearchValues.Create("\x0000\x0001\x0002\x0003\x0004\x0005\x0006\x0007\x0008\x000B\x000C\x000E\x000F\x0010\x0011\x0012\x0013\x0014\x0015\x0016\x0017\x0018\x0019\x001A\x001B\x001C\x001D\x001E\x001F");
+#else
+    private static readonly char[] AllControlChars = new char[] { '\x0000', '\x0001', '\x0002', '\x0003', '\x0004', '\x0005', '\x0006', '\x0007', '\x0008', '\x0009', '\x000A', '\x000B', '\x000C', '\x000D', '\x000E', '\x000F', '\x0010', '\x0011', '\x0012', '\x0013', '\x0014', '\x0015', '\x0016', '\x0017', '\x0018', '\x0019', '\x001A', '\x001B', '\x001C', '\x001D', '\x001E', '\x001F' };
+    private static readonly char[] NonWhitespaceControlChars = new char[] { '\x0000', '\x0001', '\x0002', '\x0003', '\x0004', '\x0005', '\x0006', '\x0007', '\x0008', '\x000B', '\x000C', '\x000E', '\x000F', '\x0010', '\x0011', '\x0012', '\x0013', '\x0014', '\x0015', '\x0016', '\x0017', '\x0018', '\x0019', '\x001A', '\x001B', '\x001C', '\x001D', '\x001E', '\x001F' };
```

Comment:
IDE0300: (NETCORE_ENGINEERING_TELEMETRY=Build) Collection initialization can be simplified